lib/internal/test_runner/utils.js

```diff
@@ -29,16 +29,10 @@ const { compose } = require('stream');
 
 const kMultipleCallbackInvocations = 'multipleCallbackInvocations';
 const kRegExpPattern = /^\/(.*)\/([a-z]*)$/;
-const kSupportedFileExtensions = /\.[cm]?js$/;
-const kTestFilePattern = /((^test(-.+)?)|(.+[.\-_]test))\.[cm]?js$/;
 
-function doesPathMatchFilter(p) {
-  return RegExpPrototypeExec(kTestFilePattern, basename(p)) !== null;
-}
+const kPatterns = ['test', 'test/**/*', 'test-*', '*[.\\-_]test'];
+const kDefaultPattern = `**/{${ArrayPrototypeJoin(kPatterns, ',')}}.?(c|m)js`;
```

Comment:
what about `.js` files? wouldn't this regex only match cjs or mjs, but not js? I think you need:
```suggestion
const kDefaultPattern = `**/{${ArrayPrototypeJoin(kPatterns, ',')}}.?[cm]?js`;
```
just like in the deleted line 33

Reply:
this is not exactly a regex - it follows https://www.gnu.org/software/bash/manual/html_node/Pattern-Matching.html

Reply:
ahhh ok, that was my confusion then.